lib/foreman_maintain/utils/service.rb

```diff
@@ -0,0 +1,14 @@
+require 'foreman_maintain/utils/service/abstract'
+require 'foreman_maintain/utils/service/systemd'
+require 'foreman_maintain/utils/service/remote_db'
+
+module ForemanMaintain::Utils
+  def self.system_service(name, priority, options = {})
+    db_feature = options.fetch(:db_feature, nil)
+    if name =~ /^(postgresql|.*mongod)$/ && db_feature && !db_feature.local?
```

Comment:
Does it make sense to use regex here or could this be an option that is passed in?

Reply:
@johnpmitsch could you please explain how it should look like and where the option should go from? I'm not sure I understand.

Reply:
i.e. (untested)
```ruby
      system_service('postgresql', 10, :component => 'foreman',
                                       :db_feature => feature(:foreman_database),
                                       :db_service => true,
```

```ruby
module ForemanMaintain::Utils
  def self.system_service(name, priority, options = {})
    db_feature = options.fetch(:db_feature, nil)
    db_service = options.fetch(:db_service, false)
    if db_service && db_feature && !db_feature.local?
      Service::RemoteDB.new(name, priority, options)
    else
      Service::Systemd.new(name, priority, options)
    end
  end
end
```
I'm not sure if I'm missing some other reason for the pattern matching, but it seems like if there is an option hash we should use that to set if its a postgres or mongo db related service. The pattern matching will probably be fine, but we don't know what kind of services will be introduced in the future and what they will be called.

Reply:
Thanks a lot for the explanation, now I see what you mean. This make sense. I wanted the factory to be black box where all the logic around choosing the right Service class is concentrated. I'm still not sure if telling the factory what the service should be is a good thing. I'll have to think about it.
Any opinion @iNecas?

Reply:
What about something like:

```ruby
if db_feature && db_feature.respond_to?(:local?) && !db_feature.local?
```

Reply:
@iNecas okay, lets give it a try. I'll update.

Reply:
@iNecas, @johnpmitsch updated, please check.